src/WebJobs.Script/OutOfProc/Http/DefaultHttpWorkerService.cs

```diff
@@ -0,0 +1,178 @@
+                {
+                    // handle http output binding
+                    scriptInvocationResult.Outputs.Add(httpOutputBinding.Name, invocationResponse);
+                    // handle $return
+                    scriptInvocationResult.Return = invocationResponse;
+                }
+                scriptInvocationContext.ResultSource.SetResult(scriptInvocationResult);
+            }
+            catch (Exception responseEx)
+            {
+                scriptInvocationContext.ResultSource.TrySetException(responseEx);
+            }
+        }
+
+        internal async Task ProcessDefaultInvocationRequest(ScriptInvocationContext scriptInvocationContext)
+        {
+            try
+            {
+                HttpScriptInvocationContext httpScriptInvocationContext = await scriptInvocationContext.ToHttpScriptInvocationContext();
+
+                // Build httpRequestMessage from scriptInvocationContext
+                HttpRequestMessage httpRequestMessage = new HttpRequestMessage();
+                AddRequestHeadersAndSetRequestUri(httpRequestMessage, scriptInvocationContext.FunctionMetadata.Name, scriptInvocationContext.ExecutionContext.InvocationId.ToString());
+                httpRequestMessage.Method = HttpMethod.Post;
+                httpRequestMessage.Content = new ObjectContent<HttpScriptInvocationContext>(httpScriptInvocationContext, new JsonMediaTypeFormatter());
+
+                _logger.LogDebug("Sending http request for function:{functionName} invocationId:{invocationId}", scriptInvocationContext.FunctionMetadata.Name, scriptInvocationContext.ExecutionContext.InvocationId);
+                HttpResponseMessage response = await _httpClient.SendAsync(httpRequestMessage);
+                _logger.LogDebug("Received http request for function:{functionName} invocationId:{invocationId}", scriptInvocationContext.FunctionMetadata.Name, scriptInvocationContext.ExecutionContext.InvocationId);
+
+                // Only process output bindings if response is succeess code
+                response.EnsureSuccessStatusCode();
+
+                HttpScriptInvocationResult invocationResult = await response.Content.ReadAsAsync<HttpScriptInvocationResult>();
+
+                ScriptInvocationResult scriptInvocationResult = new ScriptInvocationResult();
+                if (invocationResult != null)
+                {
+                    if (invocationResult.Outputs == null || !invocationResult.Outputs.Any())
+                    {
+                        _logger.LogDebug("Outputs not set on http response for invocationId:{invocationId}", scriptInvocationContext.ExecutionContext.InvocationId);
+                    }
+                    if (invocationResult.ReturnValue == null)
+                    {
+                        _logger.LogDebug("ReturnValue not set on http response for invocationId:{invocationId}", scriptInvocationContext.ExecutionContext.InvocationId);
+                    }
```

Comment:
Are these actually errors? Or are they just logging this as information? I notice you're not throwing in these checks.

Reply:
Yes, logging to make it easier for us to understand the flow. Should setting outputs be enforced?